Source/Diagnostics/ReducedDiags/BeamRelevant.cpp

```diff
@@ -0,0 +1,376 @@
+            return a * p.rdata(PIdx::w) / w_sum;
+        });
+
+        // x times ux
+        auto xux = ReduceSum( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        {
+            auto a = (p.pos(0)-x_mean) * (p.rdata(PIdx::ux)-ux_mean);
+            return a * p.rdata(PIdx::w) / w_sum;
+        });
+
+        #if (AMREX_SPACEDIM == 3)
+        // y times uy
+        auto yuy = ReduceSum( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        {
+            auto a = (p.pos(1)-y_mean) * (p.rdata(PIdx::uy)-uy_mean);
+            return a * p.rdata(PIdx::w) / w_sum;
+        });
+        #endif
+
+        // z times uz
+        auto zuz = ReduceSum( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        {
+            auto a = (p.pos(index_z)-z_mean) * (p.rdata(PIdx::uz)-uz_mean);
+            return a * p.rdata(PIdx::w) / w_sum;
+        });
+
+        // reduced sum over mpi ranks
+        ParallelDescriptor::ReduceRealSum
+            ( x_rms, ParallelDescriptor::IOProcessorNumber());
+        #if (AMREX_SPACEDIM == 3)
+        ParallelDescriptor::ReduceRealSum
+            ( y_rms, ParallelDescriptor::IOProcessorNumber());
+        #endif
+        ParallelDescriptor::ReduceRealSum
+            ( z_rms, ParallelDescriptor::IOProcessorNumber());
+        ParallelDescriptor::ReduceRealSum
+            (ux_rms, ParallelDescriptor::IOProcessorNumber());
+        ParallelDescriptor::ReduceRealSum
+            (uy_rms, ParallelDescriptor::IOProcessorNumber());
+        ParallelDescriptor::ReduceRealSum
+            (uz_rms, ParallelDescriptor::IOProcessorNumber());
+        ParallelDescriptor::ReduceRealSum
+            (gm_rms, ParallelDescriptor::IOProcessorNumber());
+        ParallelDescriptor::ReduceRealSum
+            (   xux, ParallelDescriptor::IOProcessorNumber());
+        #if (AMREX_SPACEDIM == 3)
+        ParallelDescriptor::ReduceRealSum
+            (   yuy, ParallelDescriptor::IOProcessorNumber());
+        #endif
+        ParallelDescriptor::ReduceRealSum
+            (   zuz, ParallelDescriptor::IOProcessorNumber());
+
+        // save data
+        #if (AMREX_SPACEDIM == 3)
+        m_data[0]  = x_mean;
+        m_data[1]  = y_mean;
+        m_data[2]  = z_mean;
+        m_data[3]  = ux_mean * m;
+        m_data[4]  = uy_mean * m;
+        m_data[5]  = uz_mean * m;
+        m_data[6]  = gm_mean;
+        m_data[7]  = std::sqrt(x_rms);
+        m_data[8]  = std::sqrt(y_rms);
+        m_data[9]  = std::sqrt(z_rms);
+        m_data[10] = std::sqrt(ux_rms * m);
+        m_data[11] = std::sqrt(uy_rms * m);
+        m_data[12] = std::sqrt(uz_rms * m);
+        m_data[13] = std::sqrt(gm_rms);
+        m_data[14] = std::sqrt(std::abs(x_rms*ux_rms-xux*xux)) / PhysConst::c;
+        m_data[15] = std::sqrt(std::abs(y_rms*uy_rms-yuy*yuy)) / PhysConst::c;
+        m_data[16] = std::sqrt(std::abs(z_rms*uz_rms-zuz*zuz)) / PhysConst::c;
```

Comment:
Looks good because WarpX uses `gamma*v` as the momentum in the code, for convenience (both are per particle, while `m` is per species) and performance.